packages/medusa/src/strategies/product-export.ts

```diff
@@ -0,0 +1,670 @@
+            productOption?.options[i]?.title ?? "",
+          entityName: "product",
+        })
+        .set(`Option ${i + 1} Value`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.options[i]?.value ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private appendMoneyAmountDescriptors(maxMoneyAmountCount: number): void {
+    for (let i = 0; i < maxMoneyAmountCount; ++i) {
+      this.columnDescriptors
+        .set(`Price ${i + 1} Currency code`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.currency_code?.toLowerCase() ?? "",
+          entityName: "variant",
+        })
+        .set(`Price ${i + 1} Region name`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.region?.name?.toLowerCase() ?? "",
+          entityName: "variant",
+        })
+        .set(`Price ${i + 1} Amount`, {
+          accessor: (variant: ProductVariant) =>
+            variant?.prices[i]?.amount?.toString() ?? "",
+          entityName: "variant",
+        })
+    }
+  }
+
+  private buildProductVariantLines(product: Product): string[] {
+    const outputLineData: string[] = []
+
+    for (const variant of product.variants) {
+      const variantLineData: string[] = []
+      for (const [, columnSchema] of this.columnDescriptors.entries()) {
+        if (columnSchema.entityName === "product") {
+          variantLineData.push(columnSchema.accessor(product))
+        }
+        if (columnSchema.entityName === "variant") {
+          variantLineData.push(columnSchema.accessor(variant))
+        }
+      }
+      variantLineData.push(this.NEWLINE_)
+      outputLineData.push(variantLineData.join(this.DELIMITER_))
+    }
+
+    return outputLineData
+  }
+
+  private async handleProcessingErrors(
+    batchJobId: string,
+    err: unknown,
+    {
+      offset,
+      count,
+      advancementCount,
+      progress,
+    }: {
+      offset: number
+      count: number
+      advancementCount: number
+      progress: number
+    }
+  ): Promise<void> {
+    return await this.atomicPhase_(async (transactionManager) => {
+      const batchJob = await this.batchJobService_
+        .withTransaction(transactionManager)
+        .retrieve(batchJobId)
+
+      const retryCount = batchJob.context.retry_count ?? 0
+      const maxRetry = batchJob.context.max_retry ?? this.defaultMaxRetry
+
+      if (err instanceof MedusaError) {
+        await this.batchJobService_
+          .withTransaction(transactionManager)
+          .setFailed(batchJob)
+      } else if (retryCount < maxRetry) {
+        await this.batchJobService_
+          .withTransaction(transactionManager)
+          .update(batchJobId, {
+            context: {
+              ...batchJob.context,
+              retry_count: retryCount + 1,
+              offset,
+            },
+            result: {
+              ...batchJob.result,
+              count,
+              advancementCount,
```

Comment:
comment: To be modified if other suggestion is incorporated
```suggestion
              advancement_count: advancementCount,
```